Add codespell support (config, workflow to detect/not fix) and make it fix few typos

yarikoptic opened the pull request:
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.
Per request at  https://github.com/napari/napari/pull/7619#issuecomment-2670321891

yarikoptic committed:
Add rudimentary codespell config

yarikoptic committed:
Add github action to codespell main on push and PRs

yarikoptic committed:
Add pre-commit definition for codespell

yarikoptic committed:
tuneup to codespell config

yarikoptic committed:
[DATALAD RUNCMD] Do interactive fixing of some ambigous typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 2",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

psobolewskiPhD commented on docs/community/code_of_conduct.md:
```diff
@@ -57,7 +57,7 @@ You can report issues to the napari Code of Conduct committee, at napari-conduct
 * [Kira Evans](https://github.com/kne42)
 
-If your report involves any members of the committee, or if they feel they have a conflict of interest in handling it, then they will recuse themselves from considering your report. Alternatively, if for any reason you feel uncomfortable making a report to the committee, then you can also contact:
+If your report involves any members of the committee, or if they feel they have a conflict of interest in handling it, then they will recurse themselves from considering your report. Alternatively, if for any reason you feel uncomfortable making a report to the committee, then you can also contact:
```
```suggestion
If your report involves any members of the committee, or if they feel they have a conflict of interest in handling it, then they will recuse themselves from considering your report. Alternatively, if for any reason you feel uncomfortable making a report to the committee, then you can also contact:
```
Revert please https://www.merriam-webster.com/dictionary/recuse

psobolewskiPhD commented on docs/community/code_of_conduct_reporting.md:
```diff
@@ -107,4 +107,4 @@ The committee will never publicly discuss the issue; all public statements will
 ## Conflicts of interest
 
-In the event of any conflict of interest, a committee member must immediately notify the other members, and recuse themselves if necessary.
+In the event of any conflict of interest, a committee member must immediately notify the other members, and recurse themselves if necessary.
```
```suggestion
In the event of any conflict of interest, a committee member must immediately notify the other members, and recuse themselves if necessary.
```
Revert please https://www.merriam-webster.com/dictionary/recuse

psobolewskiPhD commented on docs/plugins/advanced_topics/npe2_migration_guide.md:
```diff
@@ -276,7 +276,7 @@ function actually instantiating the widget. It accepts only one argument: a
 functionality like private methods.
 
-Similarly `napari_experimental_provide_function` hooks return ane or more
+Similarly `napari_experimental_provide_function` hooks return and or more
```
```suggestion
Similarly `napari_experimental_provide_function` hooks return one or more
```
It is a typo, but not the right fix 🤣

yarikoptic replied:
- filed https://github.com/codespell-project/codespell/pull/3645

psobolewskiPhD reviewed (commented):
Caught a few false positives.
How will those work going forward? will they get re-flagged?

General question: what do we want to do about release notes? There are typos there, but those are the titles of the PRs, which are the source of truth?
Should we skip the release notes? or fix them? fix the PR titles?

yarikoptic commented:
> How will those work going forward? will they get re-flagged?

a few ways to ignore them -(I already ignored some) - either via config (added in this PR) or "inline" (less applicable in docs) . See https://github.com/codespell-project/codespell?tab=readme-ov-file#using-a-config-file .


Let me address those false positives now, good eye @psobolewskiPhD !

yarikoptic committed:
[DATALAD RUNCMD] Do interactive fixing of some ambigous typos

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w -i 3 -C 2 ./docs/release/release_0_5_0.md",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic committed:
[DATALAD RUNCMD] run codespell throughout fixing few left typos automagically

=== Do not change lines below ===
{
 "chain": [],
 "cmd": "codespell -w",
 "exit": 0,
 "extra_inputs": [],
 "inputs": [],
 "outputs": [],
 "pwd": "."
}
^^^ Do not change lines above ^^^

yarikoptic commented:
done, pushed, also spotted/ignored one more name.

psobolewskiPhD commented:
I asked this on the napari/napari PR, but i think it's good to have the info here too.
So i'm pretty cool with this.
But can you say a few words how this will work going forward?
Say a contributor makes a PR with a typo.
What happens? pre-commit flags it and blocks merge?
What if it's a false positive like the ones I caught above?

psobolewskiPhD committed:
Merge branch 'main' into enh-codespell
